### PR TITLE
29217: Fixed -  Prefix and Suffix Translation

### LINKED ISSUE #29217: Prefix and Suffix Translation
### Preconditions (*)
1. M2.3.5-p1, 2.4-develop
2. English (default) and german store view

### Steps to reproduce (*)
1. Set "Show Prefix" to "Required" in Stores/Configuration/Customers/Customer Configuration/Name and Address Options
2. Set "Mr.;Ms." for "Prefix Dropdown Options"
3. Add a german locale file which translate "Mr." to "Herr" and "Ms." to "Frau"
"Mr.","Herr"
"Ms.","Frau"

### Expected result (*)
1. Every occurrence of the prefix should translated
image
image

### Actual result (*)
1. Only the prefix in frontend address form is translated

vendor/magento/module-customer/view/frontend/templates/widget/name.phtml
image

2. Other occurrence are not translated

image
image
image

So i have added the translation to prefix and suffix in this files: 

Magento\Customer\Helper\View::getCustomerName()
image

Magento\Customer\Block\Address\Renderer\DefaultRenderer::renderArray()
image

Magento\Customer\Model\Options::prepareNamePrefixSuffixOptions()
image

Magento\Customer\Model\Address\AbstractAddress::getName()
image

Magento\Sales\Model\Order\Address::getName()
image

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by solwininfotech**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#29217  Prefix and Suffix Translation

### Manual testing scenarios (*)

1.From admin panel, Set below option from  Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options.
	a. Show Prefix = Yes
	b. Prefix Dropdown Options = Ms.;Mr.;Mrs. (Use Semicolon (;) to separate values)
	c. Show Suffix = Yes
	d. Suffix Dropdown Options = Jr.;Sr.;MD.(Use Semicolon (;) to separate values)
2. Create another store with a different language.
For Example: Create a store german and add a german language pack for this.
Add translation words in german locale file(.csv) that translate "Mr." to "Herr" and "Ms." to "Frau" words and so on.
3. Now check below pages in frontend with different stores(Default store - English Language and German store - German language)
	a.Customer "Create an Account" page.
	b.Login as a customer and open the "My Account" page and Edit Contact Information.
	c.Add any product to the cart as a guest customer and open the checkout page.
 
Before Fix Create Account page

before_registration

After Fix Create Account page

after-registration-page

Before Fix Checkout Page

before_checkout

After Fix Checkout Page

after-checkout_page

Before Fix Account Details Edit Page

before_account-edit-page

After Fix Account Details Edit Page
after-address-edit-page

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by solwininfotech** [comment]
@magento give me test instance

**Comment by solwininfotech** [comment]
#dmcdindia2020

**Comment by solwininfotech** [comment]
@magento run all tests

**Comment by solwininfotech** [comment]
@magento give me test instance

File: app/code/Magento/Customer/Block/Address/Renderer/DefaultRenderer.php [hunk 1]
```diff
@@ -189,6 +189,9 @@ public function renderArray($addressAttributes, $format = null)
                         $data[$key] = $v;
                     }
                 }
+                if(in_array($attributeCode,['prefix','suffix'])) {
```
Comment (rogyar) [inline app/code/Magento/Customer/Block/Address/Renderer/DefaultRenderer.php]: Please, make sure you use a proper indentation for your code

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
Hi @solwininfotech. Thank you for your contribution. Please, check my comments above as well as the failing static tests. 

Thank you!

**Comment by solwininfotech** [comment]
Hi @rogyar 
Thank you for reviewing it, I am working on your sugession.

**Comment by solwininfotech** [comment]
@magento run all tests

**Comment by solwininfotech** [comment]
@magento run all tests

**Comment by solwininfotech** [comment]
@rogyar,  changes that have been done and pushed.

**Comment by rogyar** [comment]
Hi @solwininfotech. Please, check the failing static tests
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/30413/e4cd3306770880fb95b2e6a9459b2b61/Statics/allure-report-b2b/index.html#suites/51d3ab4d5d77a0bc6a7ea344e9e7e6be/352209e68147c7d7/

**Comment by solwininfotech** [comment]
@magento run Static Tests

**Comment by solwininfotech** [comment]
@magento run Static Tests

**Comment by solwininfotech** [comment]
@magento run Static Tests

**Comment by solwininfotech** [comment]
@magento run all tests

**Comment by solwininfotech** [comment]
@rogyar, Static test is passed now, I have checked for failed tests, it shows failed in other files which are not updated in this commit. 
Please help.

**Review by rogyar — APPROVED** [review]
Not sure we will be able to cover it with autotests somehow

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8391](https://jira.corp.magento.com/browse/ENGCOM-8391) has been created to process this Pull Request
:eight_spoked_asterisk: @rogyar, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1.From admin panel, go to Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options.  Set the following options:
- Show Prefix = Required
- Prefix Dropdown Options = Ms.;Mr.;Mrs. (Use Semicolon (;) to separate values)
- Show Suffix = Required
- Suffix Dropdown Options = Jr.;Sr.;MD.(Use Semicolon (;) to separate values)
name_address options
2. Install German language pack 
3. Add the following `"Mr.","Herr" "Ms.","Frau"` to the de_DE.csv file like this:
storm_de
4. Create a new Store view
5. Go to Stores - Configuration, change the Scope to the previously created Storeview
6. Change the Locale to _German (Germany)_
locale_config
7. Go to Storefront and switch to the previously created German Storeview
8. Add any product to Cart and click Proceed to Checkout 
9. Observe the _Namenspräfix_ dropdown

**Before** applying changes provided in this PR
mrms

**After** switching to the PR
mrms_after

**Pull request merged**